src/platform/plugins/shared/workflows_management/public/widgets/workflow_yaml_editor/lib/monaco_providers/template_expression_hover_provider.ts

```diff
@@ -0,0 +1,254 @@
+
+    editor.onMouseMove((e) => {
+      const currentMousePos = e.target.position;
+
+      // Clear any pending timeout
+      if (mouseMoveTimeout) {
+        clearTimeout(mouseMoveTimeout);
+        mouseMoveTimeout = null;
+      }
+
+      // Only set a timeout to clear if mouse is not over text content
+      if (e.target.type !== monaco.editor.MouseTargetType.CONTENT_TEXT) {
+        // Mouse is not over text (e.g., scrollbar, margin, etc.)
+        // Don't clear decoration immediately to prevent flicker
+      } else if (!currentMousePos) {
+        // No position means mouse is away from content
+        mouseMoveTimeout = setTimeout(() => {
+          // Timeout-based cleanup if needed
+        }, 200);
+      }
+    });
+
+    editor.onDidScrollChange(() => {
+      // Scroll change handling if needed
+    });
+  }
+
+  async provideHover(
+    model: monaco.editor.ITextModel,
+    position: monaco.Position
+  ): Promise<monaco.languages.Hover | null> {
+    try {
+      // Get execution context - return early if not available
+      const executionContext = this.getExecutionContext();
+      if (!executionContext) {
+        return null;
+      }
+
+      // Parse template expression at cursor position
+      const templateInfo = parseTemplateAtPosition(model, position);
+      if (!templateInfo || !templateInfo.isInsideTemplate) {
+        return null;
+      }
+
+      // Determine what to evaluate
+      let value: any;
+      let evaluatedPath: string;
+
+      if (templateInfo.filters.length > 0 && templateInfo.isOnFilter) {
+        // Cursor is on the filter part - evaluate with filters
+        evaluatedPath = templateInfo.expression; // Show full expression in hover title
+
+        value = evaluateExpression({
+          expression: templateInfo.expression,
+          context: executionContext,
+        });
+      } else {
+        // Cursor is on the variable path (not filter) - resolve path only
+        evaluatedPath = templateInfo.pathUpToCursor.join('.');
+
+        value = evaluateExpression({
+          expression: evaluatedPath,
+          context: executionContext,
+        });
+      }
+
+      // If value is undefined, show a message
+      if (value === undefined) {
+        return {
+          range: templateInfo.templateRange,
+          contents: [
+            {
+              value: this.formatUndefinedMessage([evaluatedPath]),
+            },
+          ],
+        };
+      }
+
+      // Format hover content
+      const hoverContent = this.formatHoverContent([evaluatedPath], value);
+
+      // Add a visible decoration to highlight the hovered segment
+      this.addHoverDecoration(templateInfo.templateRange);
+
+      return {
+        range: templateInfo.templateRange,
+        contents: [hoverContent],
+      };
+    } catch (error) {
+      return null;
+    }
+  }
+
+  /**
+   * Add a visible decoration to highlight the hovered segment
+   */
+  private addHoverDecoration(range: monaco.Range) {
```

Comment:
A separate decoration for a hover looks like an overkill, to increase contrast we can add this property in `use_workflows_monaco_theme`:
```ts
        // more contrast for hover highlight background
        'editor.hoverHighlightBackground': chroma(
          transparentize(euiTheme.colors.primary, 0.25)
        ).hex(),
``` 

pros:
 - text is not shigting on hover, similar to VS Code
 - no additional rendering/painting
 - we don't need to handle mouseleave
 - less code to maintain
 
 
https://github.com/user-attachments/assets/9498501f-a8b5-47e4-b3c8-50de677e1dc9

wdyt?